VideoLocker/src/main/java/org/edx/mobile/event/NewVersionAvailableEvent.java

```diff
@@ -0,0 +1,241 @@
+                eventBus.getStickyEvent(NewVersionAvailableEvent.class);
+        if (postedEvent == null || event.compareTo(postedEvent) > 0) {
+            eventBus.postSticky(event);
+        }
+    }
+
+    @Nullable
+    private final Version newVersion;
+    @Nullable
+    private final Date lastSupportedDate;
+    private final boolean isUnsupported;
+
+    private boolean isConsumed;
+
+    /**
+     * The logger for this class.
+     */
+    private final Logger logger = new Logger(NewVersionAvailableEvent.class);
+
+    /**
+     * Construct a new instance of NewVersionAvailableEvent. Any individual parameter can be null or
+     * false, but at least one needs to be non-null or true (and in the case of the new version
+     * number parameter, also greater than the current build's version number) in order for the
+     * event to be valid. The constructor is private because the class is only supposed to be
+     * initialized from the {@link #post(Version, Date, boolean)} method.
+     *
+     * @param newVersion        The version number of the latest release of the app.
+     * @param lastSupportedDate The last date on which the current version of the app will be
+     *                          supported.
+     * @param isUnsupported     Whether the current version is unsupported. This is based on whether
+     *                          we're getting HTTP 426 errors, and thus can't be inferred from the
+     *                          last supported date (the two properties may not be consistent with
+     *                          each other due to wrong local clock time or an inconsistency in the
+     *                          server configurations).
+     * @throws IllegalArgumentException If all of the parameters are {@code null} or {@code false}.
+     * @throws IllegalStateException    If the current build's version number doesn't correspond to
+     *                                  the schema.
+     */
+    private NewVersionAvailableEvent(@Nullable final Version newVersion,
+                                     @Nullable final Date lastSupportedDate,
+                                     final boolean isUnsupported)
+            throws IllegalArgumentException {
+        if (!isUnsupported && lastSupportedDate == null) {
+            /* If the new version number parameter was also provided as a null
+             * value, or as a value that is not greater than the current
+             * build's version number, then throw an IllegalArgumentException.
+             */
+            if (newVersion == null) {
+                throw new IllegalArgumentException(
+                        "At least one parameter needs to be non-null or true.");
+            } else {
+                final Version currentVersion;
+                try {
+                    currentVersion = new Version(BuildConfig.VERSION_NAME);
+                } catch (ParseException e) {
+                    logger.error(e);
```

Comment:
I think this should be logged to Crashlytics as well, since it would indicate broken functionality.